File: android/app/src/main/java/com/kvp/salesnavigator/FirebaseCrashlyticsPlugin.java
package com.kvp.salesnavigator;

import com.getcapacitor.JSObject;
import com.getcapacitor.Plugin;
import com.getcapacitor.PluginCall;
import com.getcapacitor.PluginMethod;
import com.getcapacitor.annotation.CapacitorPlugin;
import com.google.firebase.crashlytics.FirebaseCrashlytics;

@CapacitorPlugin(name = "FirebaseCrashlyticsPlugin")
public class FirebaseCrashlyticsPlugin extends Plugin {

    @PluginMethod
    public void setEnabled(PluginCall call) {
        boolean enabled = call.getBoolean("enabled", true);
        FirebaseCrashlytics.getInstance().setCrashlyticsCollectionEnabled(enabled);
        call.resolve();
    }

    @PluginMethod
    public void log(PluginCall call) {
        String message = call.getString("message");
        if (message != null) {
            FirebaseCrashlytics.getInstance().log(message);
        }
        call.resolve();
    }

    @PluginMethod
    public void setUserId(PluginCall call) {
        String userId = call.getString("userId");
        if (userId != null) {
            FirebaseCrashlytics.getInstance().setUserId(userId);
        }
        call.resolve();
    }

    @PluginMethod
    public void recordException(PluginCall call) {
        String message = call.getString("message");
        if (message != null) {
            FirebaseCrashlytics.getInstance().recordException(new Exception(message));
        }
        call.resolve();
    }

    @PluginMethod
    public void crash(PluginCall call) {
        call.resolve(); // Resolve before crashing so the bridge doesn't hang
        throw new RuntimeException("Test crash from app - " + call.getString("message"));
    }
}

File: ios/App/App/FirebaseCrashlyticsPlugin.swift
import Foundation
import Capacitor
import FirebaseCrashlytics

@objc(FirebaseCrashlyticsPlugin)
public class FirebaseCrashlyticsPlugin: CAPPlugin {

    @objc func setEnabled(_ call: CAPPluginCall) {
        let enabled = call.getBool("enabled") ?? true
        Crashlytics.crashlytics().setCrashlyticsCollectionEnabled(enabled)
        call.resolve()
    }

    @objc func log(_ call: CAPPluginCall) {
        if let message = call.getString("message") {
            Crashlytics.crashlytics().log(message)
        }
        call.resolve()
    }

    @objc func setUserId(_ call: CAPPluginCall) {
        if let userId = call.getString("userId") {
            Crashlytics.crashlytics().setUserID(userId)
        }
        call.resolve()
    }

    @objc func recordException(_ call: CAPPluginCall) {
        if let message = call.getString("message") {
            let error = NSError(domain: "AppError", code: 0, userInfo: [NSLocalizedDescriptionKey: message])
            Crashlytics.crashlytics().record(error: error)
        }
        call.resolve()
    }

    @objc func crash(_ call: CAPPluginCall) {
        call.resolve()
        // Trigger a crash after resolving the call
        fatalError("Test crash from app - " + (call.getString("message") ?? ""))
    }
}
